### PR TITLE
feat(header forwarding): forward headers to native node requests

### PR DESCRIPTION
**Pull request by potterbm-cb**
### Motivation

In order to support some common use cases, Mesh servers need the ability to pass request headers through to the native nodes, and to return headers from native node requests back to the caller.

Use cases:
* **passing a client ID header to the nodes** — this client ID shouldn't originate with the Mesh server since it may have many different callers, so it should be passed through to the native node requests
* **sticky session cookies** — to support sticky sessions on native nodes, AWS uses cookies. Cookies are managed by the `Cookie` and `Set-Cookie` headers, which will need to be passed from the caller to the native node, and returned from the native node ALB to the caller

### Solution

This PR adds a utility, `ContextWithHeaders` which uses `context` to inject headers into rpc requests created by `go-ethereum` (https://github.com/ethereum/go-ethereum/pull/26023)

It also adds two forms of plumbing:
1. a `contextFromRequest` method in all controllers — this allows users of `mesh-geth-sdk` to use the context to add headers to outgoing native node requests
2. middleware applied to the whole server — this allows users of `mesh-geth-sdk` to add headers to the Mesh response, possibly using a utility like `HeaderForwarder` from https://github.com/coinbase/mesh-sdk-go/pull/507

### Open questions

no open questions

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>1</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>2</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

**Review by jingweicb — APPROVED** [review]

File: headers/headers.go
Comment (marklandgrebe-cb) [inline headers/headers.go:10]: How large is the lift to filter for interesting headers before merging?

**Review by marklandgrebe-cb — DISMISSED** [review]

**Review by jingweicb — DISMISSED** [review]

**Review by marklandgrebe-cb — APPROVED** [review]

**Comment by its2badwespend** [comment]
> ### Motivation
> 
> In order to support some common use cases, Mesh servers need the ability to pass request headers through to the native nodes, and to return headers from native node requests back to the caller.
> 
> Use cases:
> * **passing a client ID header to the nodes** — this client ID shouldn't originate with the Mesh server since it may have many different callers, so it should be passed through to the native node requests
> * **sticky session cookies** — to support sticky sessions on native nodes, AWS uses cookies. Cookies are managed by the `Cookie` and `Set-Cookie` headers, which will need to be passed from the caller to the native node, and returned from the native node ALB to the caller
> 
> ### Solution
> 
> This PR adds a utility, `ContextWithHeaders` which uses `context` to inject headers into rpc requests created by `go-ethereum` (https://github.com/ethereum/go-ethereum/pull/26023)
> 
> It also adds two forms of plumbing:
> 1. a `contextFromRequest` method in all controllers — this allows users of `mesh-geth-sdk` to use the context to add headers to outgoing native node requests
> 2. middleware applied to the whole server — this allows users of `mesh-geth-sdk` to add headers to the Mesh response, possibly using a utility like `HeaderForwarder` from https://github.com/coinbase/mesh-sdk-go/pull/507
> 
> ### Open questions
> 
> no open questions
>

**Pull request merged**
